eng/ReferenceAssemblies.props

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <IsReferenceAssembly Condition="'$(IsReferenceAssembly)'=='' AND ($(MSBuildProjectFullPath.Contains('\ref\')) OR $(MSBuildProjectFullPath.Contains('/ref/')))">true</IsReferenceAssembly>
+
+    <!-- Create a common root output directory for all reference assemblies -->
+    <ReferenceAssemblyOutputPath Condition="'$(ReferenceAssemblyOutputPath)' == ''">$(BaseOutputPath)ref/</ReferenceAssemblyOutputPath>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(IsReferenceAssembly)'=='true'">
+    <OutputPath>$(ReferenceAssemblyOutputPath)$(AssemblyName)/$(AssemblyVersion)/$(TargetOutputRelPath)</OutputPath>
+    <IntermediateOutputPath>$(RootIntermediateOutputPath)ref/$(AssemblyName)/$(AssemblyVersion)/$(TargetOutputRelPath)</IntermediateOutputPath>
+
+    <!-- When using Sdk-style projects, Microsoft.Common.CurrentVersion.targets gets imported before
```

Comment:
It feels like we should be able to eliminate this hack now and correctly these up in Directory.Build.props file now and also allow us to import it in the props file instead of the targets file. Can you help me bettern understandy why we cannot do that?

Reply:
I was able to delete these hacks, the only thing that I need to still override is OutDir in order for it to work and being able to import this directly in the props file.

Also, when imported directly in Directory.Build.props since AssemblyVersion is not defined, the path will no longer contain the AssemblyVersion, which shouldn't matter since we don't produce multiple versions of them.